rules/jsll/src/rules/jsll-validate-config/jsll-validate-config.ts

```diff
@@ -0,0 +1,130 @@
+/**
+ * @fileoverview Validate the required config properties.
+ */
+
+import { Category } from 'sonarwhal/dist/src/lib/enums/category';
+import { RuleContext } from 'sonarwhal/dist/src/lib/rule-context';
+import { IRule, IRuleBuilder, IScriptParse, IScanEnd } from 'sonarwhal/dist/src/lib/types';
+import { debug as d } from 'sonarwhal/dist/src/lib/utils/debug';
+
+import { isPotentialInitScript, validateNodeProps, configProps } from '../validator';
+import { isObject } from '../utils';
+
+const debug: debug.IDebugger = d(__filename);
+
+/*
+ * ------------------------------------------------------------------------------
+ * Public
+ * ------------------------------------------------------------------------------
+ */
+const generateScript = (callInitCode) => {
+    const script = `(function stubAwaInit() {
+        var originalInit = window.awa && window.awa.init ? window.awa.init.bind(window.awa) : null;
+        var configUsed;
+
+        newInit = function () {
+            configUsed = Array.prototype.slice.call(arguments);
+        };
+
+        window.awa = { init: newInit };
```

Comment:
Here you are overriding the whole `ava` object so even if you restore the original `init` later it will be different. I don't know if `awa` exposes other methods, but they will be lost.

Maybe do  `window.awa.init = newInit;`